tests/test-deploy-ad-server.sh

```diff
@@ -65,12 +74,31 @@ echo
 [ $rc -eq 0 ] || _error "Error: samba ad did not become reachable"
 
 
-
-AD_POD_IP=$(${KUBECTL_CMD} get pod -o jsonpath='{ .items[*].status.podIP }')
+AD_POD_IP=$(${KUBECTL_CMD}  get pod -o json \
+	| ${JQ_CMD} -c -M '.items[] | .metadata.name + " " + .status.podIP' \
+	| grep samba-ad-server \
+	| tr -d "\"" \
+	| awk '{print $2}')
 [ $? -eq 0 ] || _error "Error getting ad server pod IP"
-
 echo "AD pod IP: ${AD_POD_IP}"
 
+# when running over OpenShift need to patch dns-operator
+${KUBECTL_CMD} get deployment dns-operator \
+	-n openshift-dns-operator > /dev/null 2>&1
+OPENSHIFT_DNS="$?"
+if [ ${OPENSHIFT_DNS} -eq 0 ]; then
+	PATCHFILE=$(mktemp).json
+	sed -e "s/AD_SERVER_IP/${AD_POD_IP}/g" \
+		${OPENSHIFT_DNS_SNIPPET} > "${PATCHFILE}"
+
+	${KUBECTL_CMD} patch dns.operator default \
+		--type=merge --patch-file "${PATCHFILE}"
+
+	[ $? -eq 0 ] || _error "Failed patching dns-operator with $PATCHFILE"
+	unlink "${PATCHFILE}"
+	exit 0
```

Comment:
Nitpick: I missed this the first readthrough and for a moment thought the code was going to run this if-block AND the regular non-openshift approach to patching the coredns config.  I won't ask for it in this PR, but next time we need to do non-trivial work to this script it might be time to start using (more) functions,